### ISSUE TITLE
Can't use "t" in the pause menu twice in a row

### ISSUE THREAD

**Issue by DestyNova** [kstaats/karoo_gp#12]
If I run in interactive mode (I've also seen it at the end of a "minimal" mode run), then I can use the pause menu at most once per generation:

```
  2 trees [ 1 38] offer the highest fitness scores.

         (pause) t
...
 Confusion matrix:
[[3072  984]
 [2997  947]]

         (pause) t
         Select from the options given. Try again ...

         (pause) ?
         Select from the options given. Try again ...

         (pause) p
         Select from the options given. Try again ...

         (pause) 30
         Select from the options given. Try again ...
```

I'm guessing it's some kind of parsing bug where the state of some buffer is left inconsistent after the first "t" operation. All I can do now is hit enter and let the next generation run, after which I'll be able to use "t" on (at most) one of the winners from that generation.

**Comment by ByteSumoLtd** [comment]
@DestyNova  -- fyi if you are playing with the code. I have fork too and have been been playing around with some minor changes.  https://github.com/ByteSumoLtd/karoo_gp/commits/master

**Comment by kstaats** [comment]
Let's jump on the phone to discuss. A few things I want to improve, just 
no time :)

**Comment by kstaats** [comment]
To DestyNova's original issue, I offer my sincere apology for a lack of follow-through. My research in an totally different arena has dominated my time for several months. Now, my work at LIGO is bringing me back to SIMOC again, with a need to not only look into an correct these known, outstanding issues, but also bring Karoo up to Python 3.x.

I appreciate your patience, and look forward to providing an improved product soon.

**Comment by kstaats** [comment]
The final Python 2.7 version of Karoo has been pushed to github. I am now returning to active development of Karoo. The next task is to improve the means by which I present and execute the functions of the Pause menu, so as to remove the potential for this loop. I will look at ByteSumo's mods and implement soon. Stay tuned!

**Comment by kstaats** [comment]
This is fully resolved with the new Karoo GP 2.3 for Python 3.6. The menu functions are fully rebuilt with no nested loops. Please sync with the latest version and we'll call this closed! :)
